src/components/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,82 @@
+import Input from 'components/Input';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { PasswordStrength } from 'types/generic.types';
+import getPasswordStrength from 'util/passwordHelper';
+
+type PasswordInputProps = {
+  value: string;
+  onBlur: () => void;
+  error: string;
+  onChange: (value: string) => void;
+  showVisibilityIcon: boolean;
+  showPassword: boolean;
+  showPasswordStrength: boolean;
+  handleClickShowPassword: () => void;
+  label: string;
+  floatingLabels: boolean;
+};
+
+const PasswordInput = ({
+  value,
+  onBlur,
+  error,
+  onChange,
+  showVisibilityIcon,
+  showPasswordStrength,
+  showPassword,
+  handleClickShowPassword,
+  label = 'Password',
+  floatingLabels = true
+}: PasswordInputProps) => {
+  const { t } = useTranslation();
```

Comment:
all the hooks and both methods from here are good candidates to be extracted to a custom hook. please create a hook named something like `usePasswordInputUtils.ts` and extract everything there. it will improve the readability of this component